File: test/dataset/dataset_indexfile_read.cpp
#include "dataset_indexfile_reader.h"
#include "print_dataset_index_item.h"
#include "posix_file_system.h"
#include <iostream>
#include <string>
#include <memory>

using namespace std;

int main(int argc, char *argv[])
{
    string filename = "test.index";
    string partition_path = "test.part";

    if (argc == 2) {
        filename = argv[1];
    }

    auto file_system = make_shared<PosixFileSystem>();
    DatasetIndexfileReader reader(file_system, filename, partition_path);

    for (auto item: reader) {
        cout << item << "\n";
    }

    return 0;
}

File: export_py_debug/dataset_indexfile_read.py
#!/usr/bin/env python

import sys
import st_dataset

filename = "test.index"
partition_path = "test.part"

if len(sys.argv) == 2:
    filename = sys.argv[1]

print("filename:", filename)

file_system = st_dataset.create_posix_file_system()
reader = st_dataset.DatasetIndexfileReader(file_system, filename, partition_path)

for item in reader:
    print(item.file_path, item.partition_path, item.offset, item.file_size)

